src/App.tsx

```diff
@@ -1,14 +1,68 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState, useMemo } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { TodoFilterTypes } from './types/TodoFilterTypes';
+
+const getFilteredTodos = (
+  todos: Todo[],
+  query: string,
+  selectedFilter: string,
+) => {
+  let todosClone = [...todos];
+
+  if (selectedFilter !== TodoFilterTypes.All) {
+    todosClone = todosClone.filter(({ completed }) => {
+      switch (selectedFilter) {
+        case TodoFilterTypes.Active:
+          return !completed;
+
+        case TodoFilterTypes.Completed:
+          return completed;
+
+        default:
+          throw new Error('Something went wrong :(');
+      }
+    });
+  }
+
+  if (query) {
+    todosClone = todosClone.filter(({ title }) => {
+      return title.toLowerCase().includes(query.toLowerCase());
+    });
+  }
+
+  return todosClone;
+};
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
```

Comment:
Better to set `false` as a default value and change it to `true` in `useEffect` before make API call